### PR TITLE
Browserify setup and refactor to accomodate.

### PR DESCRIPTION
**Pull request by frankmariette**
## Changes

Removed RequireJS
Added and configured Browserify
Restructured application to work better with browserify and future build processes
## Why is this needed?

Bower doesn't have a standard to follow for module publishing and so using RequireJS becomes impractical to maintain. Browserify utilizes the CommonJS standard and allows usage of NPM modules in the browser without having to convert them. This opens up the usage of substantially more packages and also provides consistency in how those packages are setup.

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -43,11 +56,26 @@ gulp.task('css', function(){
 				return "Error: " + error.message;
 			})))
 		.pipe(autoprefix('last 2 version'))
-		.pipe(gulp.dest('./public/stylesheets'));
+		.pipe(gulp.dest('dist/stylesheets'));
 });
 
 gulp.task('watch', function(){
 	gulp.watch(config.sassPath + '/**/*.scss', ['css']);
+	var watcher  = watchify(browserify({
```
Comment (Felttrip) [inline Gulpfile.js:64]: Nothing of value to say I just think all the `*ify` stuff is funny.

**Comment by Felttrip** [comment]
Looks reasonable to me, I haven't built it yet, but I assume you won't need to update the readme because it's all abstracted away with gulp?

**Comment by frankmariette** [comment]
Yeah, nothing will change in how the project is setup with this. Basically trying to keep everything the same in terms of build process as much as I can to make it as pain free as possible. I mostly modified the 'watch' task that already existed and removed an old task.

File: Gulpfile.js (hunk 1 above)
Comment (ryan-endacott) [inline Gulpfile.js:64]: Lol that's great.

**Comment by ryan-endacott** [comment]
I don't know anything about gulp, but it all looks good to me.

**Comment by frankmariette** [comment]
Thanks, going to merge it in then.

**Pull request merged**